app/code/Magento/Backend/Model/Url.php

```diff
@@ -366,6 +366,20 @@ protected function _getMenu()
         return $this->_menu;
     }
 
+    /**
+     * Set scope entity
+     *
+     * @param mixed $scopeId
+     * @return \Magento\Framework\UrlInterface
+     */
+    public function setScope($scopeId)
+    {
+        parent::setScope($scopeId);
+        $this->setData('scope', $this->_scopeResolver->getScope($scopeId));
```

Comment:
Sorry, @Nazar65 , probably I made a bit confusing statement. I meant that it would be good to call parent `setScope` function so that `scope` data field is set. There is no need to set `scope` DataObject key twice as it's done in the parent function. 

This line can be removed

Reply:
ok